tests/test_utils/logging_setup.py

```diff
@@ -44,11 +46,16 @@ def setup_logging():
         "%(levelname)s - %(request)s - %(message)s"
     )
 
-    handler = ToTwistedHandler()
+    to_twisted_handler = ToTwistedHandler()
     formatter = logging.Formatter(log_format)
-    handler.setFormatter(formatter)
-    handler.addFilter(LoggingContextFilter())
-    root_logger.addHandler(handler)
+    to_twisted_handler.setFormatter(formatter)
+    to_twisted_handler.addFilter(LoggingContextFilter())
```

Comment:
I *think* this is redundant, as of #8051. (If it's not redundant, we need to do it for the other handler too.)

Reply:
Alright. I propose to test if this is redundant by
* removing filter here
* changing LoggingContextFilter to have some obvious side effect
* trigger the new log-to-stderr behaviour, and look for the side effect

Reply:
Appears to be needed. Otherwise the call to `debug` in synapse/metrics/__init__.py:596 will fail because the record doesn't have a "request" field.